src/main/java/org/jboss/modules/ModuleLoader.java

```diff
@@ -239,6 +239,19 @@ public String toString() {
         return String.format("%s@%x for finders %s", getClass().getSimpleName(), hashCode(), Arrays.toString(finders));
     }
 
+    /**
+     * Get a string representation of the given module, used in debug output and stack traces.  This method may be
+     * overridden to provide a more detailed description.  By default it returns a string consisting of the module's
+     * name and version (if any).
+     *
+     * @param module the module to describe
+     * @return the description string
+     */
+    public String getModuleDescription(final Module module) {
+        final Version version = module.getVersion();
+        return version == null ? module.getName() : module.getName() + "@" + version;
```

Comment:
I like this change. This new feature is very helpful to have module names in stack traces.
I already had an opportunity to see that feature in practice while debugging your ModuleIdentifier deprecation changes ;)